packages/tools/client-debugger/src/Registry.ts

```diff
@@ -45,6 +46,122 @@ export interface FluidClientDebuggerProps {
 	containerNickname?: string;
 }
 
+/**
+ * @internal
+ */
+export interface DebuggerRegistryEvents extends IEvent {
+	/**
+	 * Emitted when a {@link FluidClientDebugger | clientDebugger} is registered.
+	 *
+	 * @eventProperty
+	 */
+	(
+		event: "debuggerRegistered",
+		listener: (containerId: string, clientDebugger: IFluidClientDebugger) => void,
+	): void;
+
+	/**
+	 * Emitted when a {@link FluidClientDebugger | clientDebugger} is closed.
+	 *
+	 * @eventProperty
+	 */
+	(event: "debuggerClosed", listener: (containerId: string) => void): void;
+}
+
+/**
+ * Contract for maintaining a global client debugger registry to store all registered client debugger and current
+ * displaying debugger.
+ * @internal
+ */
+export class DebuggerRegistry extends IEventProvider<DebuggerRegistryEvents> {
+	/**
+	 * A map of registered debuggers, which debugger monitor one container.
+	 */
+	private readonly registeredDebuggers: Map<string, IFluidClientDebugger> = new Map();
+
+	/**
+	 * The debugger is currently displaying.
+	 */
+	private currentDisplayDebugger: IFluidClientDebugger;
```

Comment:
This library really shouldn't know anything about our visualizer library. The visualizer code should be what manages the "current" debugger. This class should just fire off its 2 change events when appropriate, and our visualizer library should respond accordingly.

Reply:
Removed it